sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts

```diff
@@ -262,13 +267,19 @@ export abstract class SFTPIntegrationBase<DownloadType> {
   }
 
   /**
-   * Delete a file from SFTP.
+   * Archives a file on SFTP to the "Archive\" directory in the original file folder.
    * @param remoteFilePath full remote file path with file name.
    */
-  async deleteFile(remoteFilePath: string): Promise<void> {
+  async archiveFile(remoteFilePath: string): Promise<void> {
     const client = await this.getClient();
     try {
-      await client.delete(remoteFilePath);
+      // Extract the directory from the file path.
+      const directoryPath = path.dirname(remoteFilePath);
+      const fileBaseName = path.basename(remoteFilePath);
+      await client.rename(
+        remoteFilePath,
+        directoryPath + ARCHIVE_DIRECTORY + fileBaseName,
```

Comment:
I would say that building of the archive directory must happen at the integration service level. Otherwise it would be very tightly coupled here. 

e.g. if any of the integration needs to have archive directory in a different name then we could not make changes specific to that integration.

Reply:
The method could be called as renameFile or moveFile with respect to previous comment.